src/hb-aat-layout-trak-table.hh

```diff
@@ -39,11 +39,15 @@ namespace AAT {
 
 struct TrackTableEntry
 {
-  inline bool sanitize (hb_sanitize_context_t *c) const
+  inline bool sanitize (hb_sanitize_context_t *c, const void *base, uint16_t size) const
   {
     TRACE_SANITIZE (this);
-    /* XXX Sanitize values */
```

Comment:
Any special thing here?

Reply:
The value array is not sanitized...

I'll try to clean up sanitize in trak soon.

In the future, it would be much better to first write the table structs and finish sanitize, before moving on to actually using the struct. The time the sanitize is written is the best time to make sure all memory access is accounted for correctly. Failing that, an XXX comment is to be added, to make sure we fix things before calling into that code. It's much harder to audit code after the sanitize is written and checked in without an XXX.

Reply:
That makes sense. For here, isn't `((base+values).sanitize (c, size))` added below enough?

Reply:
Humm. The code in front of me is this:
```
struct TrackTableEntry 
{ 
  inline bool sanitize (hb_sanitize_context_t *c) const 
  { 
    TRACE_SANITIZE (this); 
    /* XXX Sanitize values */ 
    return_trace (c->check_struct (this)); 
  } 
 
  inline float get_value (const void *base, unsigned int index) const 
  { 
    return (base+values)[index]; 
  } 
 
```

Reply:
!!! This is before the change, it should followed by `get_track_value` also.

Reply:
Sorry I thought you merged already.